pkg/operators/cli/clioperator.go

```diff
@@ -0,0 +1,173 @@
+
+func (o *cliOperator) Name() string {
+	return "cliOperator"
+}
+
+func (o *cliOperator) Init(params *params.Params) error {
+	return nil
+}
+
+func (o *cliOperator) GlobalParamDescs() params.ParamDescs {
+	return nil
+}
+
+func (o *cliOperator) InstantiateDataOperator(ctx operators.GadgetContext) (operators.DataOperatorInstance, error) {
+	return &cliOperatorInstance{
+		outputDefaults: make(map[string]string),
+		headers:        make(map[string]string),
+		mode:           ModeJSON,
+	}, nil
+}
+
+type cliOperatorInstance struct {
+	outputDefaults map[string]string
+	headers        map[string]string
+	mode           string
+}
+
+func (c *cliOperatorInstance) Name() string {
+	return "cliOperatorInstance"
+}
+
+func (c *cliOperatorInstance) Priority() int {
+	return Priority
+}
+
+func (c *cliOperatorInstance) ParamDescs(gadgetCtx operators.GadgetContext) params.ParamDescs {
+	// output format should be:
+	// --fields datasource:comma,separated,fields
+	fields := &params.ParamDesc{
+		Key:          ParamFields,
+		DefaultValue: "value",
+		Description:  "",
+	}
+
+	// Fetch defaults
+	var str strings.Builder
+	for _, v := range c.outputDefaults {
+		str.WriteString(v)
+		str.WriteByte(';')
+	}
+
+	mode := &params.ParamDesc{
+		Key:            ParamMode,
+		DefaultValue:   ModeJSON,
+		Description:    "output mode",
+		PossibleValues: []string{ModeJSON, ModeColumns},
+	}
+
+	return params.ParamDescs{fields, mode}
+}
+
+func (c *cliOperatorInstance) Prepare(gadgetCtx operators.GadgetContext, params *params.Params) error {
+	for _, ds := range gadgetCtx.GetDataSources() {
+		gadgetCtx.Logger().Debugf("subscribing to %s", ds.Name())
+
+		p, err := ds.Parser()
+		if err != nil {
+			gadgetCtx.Logger().Debugf("failed to get parser: %v", err)
+			continue
+		}
+
+		defCols := p.GetDefaultColumns()
+		gadgetCtx.Logger().Debugf("default fields: %s", defCols)
+		formatter := p.GetTextColumnsFormatter()
+
+		// Build output default
+		var flag strings.Builder
+		flag.WriteString(ds.Name())
+		flag.WriteString(":")
+		flag.WriteString(strings.Join(defCols, ","))
+		c.outputDefaults[ds.Name()] = flag.String()
+
+		// xformatter := json.NewFormatter[datasource.DataTuple](p.GetColumns().(columns.Columns[datasource.DataTuple]).GetColumnMap())
+		// p.SetEventCallback(xformatter.FormatEntries)
+
+		if c.mode == ModeColumns {
+			c.headers[ds.Name()] = fmt.Sprintf("%s\n", formatter.FormatHeader())
+		}
+
+		formatter.SetEventCallback(func(s string) {
+			fmt.Print(s)
+			fmt.Print("\n")
+		})
+
+		p.SetEventCallback(formatter.EventHandlerFunc())
+		handler, ok := p.EventHandlerFunc().(func(data *datasource.DataTuple))
+		if !ok {
+			gadgetCtx.Logger().Warnf("invalid data format: expected func(data *datasource.DataTuple), got %T",
+				p.EventHandlerFunc())
+			continue
+		}
+
+		jsonFormatter := json.New(ds)
+
+		ds.Subscribe(func(ds datasource.DataSource, data datasource.Data) error {
+			if false {
+				handler(datasource.NewDataTuple(ds, data))
+			}
```

Comment:
This is a wrapper to make `Data` compatible with the columns library / parser helpers. We should remove and simplify it once we get rid of the legacy gadgets. This just helps keeping the same codebase for both for now.

Reply:
Ah, sorry - you're right, I misunderstood. Right now, the CLI operator is always outputting JSON, while this specific code would be for outputting columns (and just isn't active because of an issue with evaluating the params). I'll push an update later on which fixes the params and adds this code back again.